File: Problems/0390-elimination-game.py
class Solution(object):
    def lastRemaining(self, n):
        """
        :type n: int
        :rtype: int
        """
        head = 1
        step = 1
        left = True

        while n > 1:
            if left or n & 1:
                head += step
            step <<= 1
            n >>= 1
            left = not left

        return head

File: Problems/0390-elimination-game.cpp
#include <bits/stdc++.h>
using namespace std;

// LeetCode solution starts from here
class Solution {
public:
    int lastRemaining(int n) {
        int head = 1;
        int step = 1;
        bool left = true;

        while (n > 1) {
            if (left || (n & 1))
                head += step;

            step <<= 1;
            n >>= 1;
            left = !left;
        }

        return head;
    }
};